frontend/src/components/Navbar.js

```diff
@@ -149,7 +114,32 @@ const NavHeader = observer(({ drawerOpen, handleDrawerOpen, drawerWidth }) => {
         <IconButton onClick={handleProfileMenuOpen} color="inherit">
           <AccountCircle />
         </IconButton>
-        <AccountMenu />
+        <Menu
+          anchorEl={anchorEl}
+          anchorOrigin={{ vertical: "top", horizontal: "right" }}
+          transformOrigin={{ vertical: "top", horizontal: "right" }}
+          open={isMenuOpen}
+          onClose={handleProfileMenuClose}
+        >
+          {rootStore.authStore.isAuthenticated ? (
+            <div>
+              <MenuItem onClick={handleProfileMenuClose}>Profile</MenuItem>
+              <MenuItem
+                onClick={() => {
+                  handleProfileMenuClose()
+                  rootStore.authStore.logout()
+                  history.push("/")
```

Comment:
nit: Prefer not to have inline functions doing multiple things id suggest moving the
 ```
rootStore.authStore.logout()
history.push("/");
```
into the `handleProfileMenuClose` so that its all encapsulated in one function

Reply:
That would cause the user to get logged out whenever the menu closes. Maybe you want a new function called logoutUserAndCloseMenu or something like that?

Reply:
youre right. actually for logout this is fine.